packages/token-bridge-sdk/src/hooks/useBalance.ts

```diff
@@ -0,0 +1,41 @@
+import { JsonRpcProvider } from '@ethersproject/providers'
+import { useBalanceContext } from '../context/balanceContext'
+import { BigNumber } from 'ethers'
+import useSWR from 'swr'
+
+const useBalance = ({
+  provider,
+  walletAddress
+}: {
+  provider: JsonRpcProvider
+  walletAddress: string
+}) => {
+  const [allBalances, setBalances] = useBalanceContext()
+  const balances = allBalances[walletAddress]?.[provider.network?.chainId] || {
+    eth: BigNumber.from(0)
+  }
+
+  const { mutate } = useSWR(
+    walletAddress && provider.network?.chainId
+      ? [walletAddress, provider.network.chainId]
+      : null,
```

Comment:
What exactly does this ternary do?

Reply:
`useSWR` being a hook, we can't call it conditionally, so the recommended way to do it, it's to pass `null` as a key. Here, we only start fetching once `walletAddress` and `provider.network.chainId` are both defined.

Reply:
Makes sense. If we don't need to check for network being undefined (the ethers types actually don't cover this) then we can remove the ternary?

Reply:
When we refresh, `walletAddress` is briefly undefined, but I moved the check to the fetch function, it should be easier to read now